includes/templates/template_default/jscript/zen_addr_pulldowns.php

```diff
@@ -11,58 +11,78 @@
  * @version $Id: Zen4All 2019 Jun 03 Modified in v1.5.7 $
  */
 // -----
-// If the site has NOT enabled states to be displayed as dropdowns, nothing to be done here!
+// If the site does NOT require a 'State' entry in an address, nothing to be done here!
 //
-if (ACCOUNT_STATE_DRAW_INITIAL_DROPDOWN !== 'true') {
+if (ACCOUNT_STATE !== 'true') {
     return;
 }
-?>
-<script>
-jQuery(document).ready(function() {
-<?php
+
 // -----
-// Create a JSON-encoded array of countries-to-zones for use by the jQuery section.
+// If the current site is using the state dropdowns, create the array that identifies the various
+// zones for the currently-active countries.  When state dropdowns aren't being used, the
+// empty $c2z array will 'instruct' the jQuery section to presume that no countries have
+// associated zones.
 //
-$countries = $db->Execute(
-    "SELECT DISTINCT zone_country_id
-       FROM " . TABLE_ZONES . "
-            INNER JOIN " . TABLE_COUNTRIES . "
-                ON countries_id = zone_country_id
-               AND status = 1
-   ORDER BY zone_country_id"
-);
-
 $c2z = [];
-foreach ($countries as $next_country) {
-    $current_country_id = $next_country['zone_country_id'];
-    $c2z[$current_country_id] = [];
-
-    $states = $db->Execute(
-        "SELECT zone_name, zone_id
+if (ACCOUNT_STATE_DRAW_INITIAL_DROPDOWN === 'true') {
+    // -----
+    // If the current site has at least one country enabled that uses zones, a JSON-encoded array of
+    // countries-to-zones will be created for use by the jQuery section.
+    //
+    $countries = $db->Execute(
+        "SELECT DISTINCT zone_country_id
            FROM " . TABLE_ZONES . "
-          WHERE zone_country_id = $current_country_id
-       ORDER BY zone_name"
+                INNER JOIN " . TABLE_COUNTRIES . "
+                    ON countries_id = zone_country_id
+                   AND status = 1
+       ORDER BY zone_country_id"
     );
-    foreach ($states as $next_state) {
-        $c2z[$current_country_id][$next_state['zone_id']] = $next_state['zone_name'];
+    foreach ($countries as $next_country) {
+        $current_country_id = $next_country['zone_country_id'];
+        $c2z[$current_country_id] = [];
+
+        $states = zen_get_country_zones($current_country_id);
+        foreach ($states as $next_state) {
+            $c2z[$current_country_id][$next_state['id']] = $next_state['text'];
+        }
     }
 }
-
-if (count($c2z) !== 0) {
+?>
+<script>
+jQuery(document).ready(function() {
+<?php
     echo '    var country_zones = \'' . addslashes(json_encode($c2z)) . '\';' . PHP_EOL;
```

Comment:
Since the creation of "country_zones" is in php and "hidden" from the IDE, it moans about unresolved variable.
Is there some reason why this is better in php, than javascript with embedded php?

Reply:
Nope, the original implementation had either an empty string or the `var` statement.  Now that the `var` is always set &mdash; just in case a site has no countries with zones &mdash; that change is possible.

Reply:
Noting, too, that I've changed this to be a `const` instead of `var`.